macro/run_trac_its.C

```diff
@@ -11,81 +11,86 @@
-        // Initialize logger
-        FairLogger *logger = FairLogger::GetLogger();
-        logger->SetLogVerbosityLevel("LOW");
-        logger->SetLogScreenLevel("INFO");
-
-        // Input and output file name
-        std::stringstream inputfile, outputfile, paramfile;
-        inputfile << "AliceO2_" << mcEngine << ".clus_" << nEvents << "_event.root";
-        paramfile << "AliceO2_" << mcEngine << ".params_" << nEvents << ".root";
-        outputfile << "AliceO2_" << mcEngine << ".trac_" << nEvents << "_event.root";
-
-        // Setup timer
-        TStopwatch timer;
-
-        // Setup FairRoot analysis manager
-        FairRunAna * fRun = new FairRunAna();
-        FairFileSource *fFileSource = new FairFileSource(inputfile.str().c_str());
-        fRun->SetSource(fFileSource);
-        fRun->SetOutputFile(outputfile.str().c_str());
-
-        // Setup Runtime DB
-        FairRuntimeDb* rtdb = fRun->GetRuntimeDb();
-        FairParRootFileIo* parInput1 = new FairParRootFileIo();
-        parInput1->open(paramfile.str().c_str());
-        rtdb->setFirstInput(parInput1);
-
-        // Setup tracker
-        Int_t n=1;            // Number of threads
-        Bool_t mcTruth=kTRUE; // kFALSE if no comparison with MC is needed
-        o2::ITS::CookedTrackerTask *trac = new o2::ITS::CookedTrackerTask(n,mcTruth);
-	trac->setContinuousMode(rate>0.);
-	
-        fRun->AddTask(trac);
-
-        fRun->Init();
-
-        o2::field::MagneticField* fld = (o2::field::MagneticField*)fRun->GetField();
-      	if (!fld) {
-      	  std::cout << "Failed to get field instance from FairRunAna" << std::endl;
-      	  return;
-      	}
-      	trac->setBz(fld->solenoidField()); //in kG
-
-        timer.Start();
-        fRun->Run();
-
-        std::cout << std::endl << std::endl;
-
-        // Extract the maximal used memory an add is as Dart measurement
-        // This line is filtered by CTest and the value send to CDash
-        FairSystemInfo sysInfo;
-        Float_t maxMemory=sysInfo.GetMaxMemory();
-        std::cout << "<DartMeasurement name=\"MaxMemory\" type=\"numeric/double\">";
-        std::cout << maxMemory;
-        std::cout << "</DartMeasurement>" << std::endl;
-
-        timer.Stop();
-        Double_t rtime = timer.RealTime();
-        Double_t ctime = timer.CpuTime();
-
-        Float_t cpuUsage=ctime/rtime;
-        cout << "<DartMeasurement name=\"CpuLoad\" type=\"numeric/double\">";
-        cout << cpuUsage;
-        cout << "</DartMeasurement>" << endl;
-        cout << endl << endl;
-        cout << "Macro finished succesfully." << endl;
-
-        std::cout << endl << std::endl;
-        std::cout << "Output file is "    << outputfile.str() << std::endl;
-        //std::cout << "Parameter file is " << parFile << std::endl;
-        std::cout << "Real time " << rtime << " s, CPU time " << ctime
-                  << "s" << endl << endl;
+void run_trac_its(Int_t nEvents = 10, TString mcEngine = "TGeant3", float rate = 0.)
+{
```

Comment:
This hard-coded input file schema is not compatible with o2sim output. In the large PR https://github.com/AliceO2Group/AliceO2/pull/862 related to TPC-ITS matching I've modified this macro to take explicitly input files real names as arguments. For backward compatibility I left this old style interface (it just composes the filenames and calls new method: https://github.com/shahor02/AliceO2/blob/dbc582b88e51567269348fa444e2fe60a55e06e8/macro/run_trac_its.C, but I am afraid some conflicts will need to be solved to make it compatible with this one.

Reply:
@shahor02  No problem. I will resolve these conflicts,  and will add to this PR a better interface to the vertexer.